### PR TITLE
[wip] Pre-releases having precedence over releases

### PR DESCRIPTION
**Pull request by podhmo**
#5

### CODE REVIEW

File: semver/tests/test_regression.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    # https://github.com/podhmo/python-semver/issues/5
+    ["<=1.2", "1.2.0", ["1.1.1", "1.2.0-pre", "1.2.0", "1.1.1-111", "1.1.1-21"]]
+]
+
+
+@pytest.mark.parametrize("op, wanted, cands", cands)
+def test_it(op, wanted, cands):
+    from semver import max_satisfying
+    got = max_satisfying(cands, op, loose=True)
+    assert got == "1.2.0"
```
Comment (memsharded) [inline semver/tests/test_regression.py]: ``assert got == wanted`` ?

Comment (podhmo) [reply semver/tests/test_regression.py]: ah yes,

**Comment by podhmo** [comment]
i see. `<=1.2` is treated as `<SemVer Comparator "<=1.2.0-0">`. so 1.2 is rejected.

**Comment by podhmo** [comment]
some test are broken, need to fix

**Comment by memsharded** [comment]
Sure, no worries. I have tried to have a look to your fix proposal, seems you are also doing other improvements and I couldn't follow, sorry I couldn't help further. 

Just another check: when you compare with ``<=1.2.0`` explicit, the prerelease ``-0`` is not added in that case, and it works

```python
wanted = "1.2" 
cands =  ["1.1.1", "1.2", "1.1.1-111", "1.1.1-21"]
result = max_satisfying(cands, "<=1.2.0", loose=True)
assert wanted == result, "result is %s" % result #Works
```

**Comment by podhmo** [comment]
tests are passed.
(and implementation is changed, for catch-up current node's implementation)

not impelemented features

- intersects()
- diff()
- min_satisifying89

**Comment by podhmo** [comment]
These tests are passed, so original issue is solved, maybe.
https://github.com/podhmo/python-semver/pull/8/files#diff-4841ac9d81b28247dd3e303936222e7b

**Pull request merged**